File: app/src/main/java/com/pgssoft/mvvm/views/adapters/BindingViewHolder.java
package com.pgssoft.mvvm.views.adapters;

import android.databinding.ViewDataBinding;
import android.support.v7.widget.RecyclerView;

import com.pgssoft.mvvm.viewmodels.BaseListItemViewModel;

/**
 * Created by bstokrocki on 07.02.2017.
 */

public class BindingViewHolder<TViewModel extends BaseListItemViewModel>
        extends RecyclerView.ViewHolder {
    private ViewDataBinding binding;
    private TViewModel viewModel;

    public BindingViewHolder(ViewDataBinding binding, TViewModel viewModel, int variable) {
        super(binding.getRoot());

        this.binding = binding;
        this.viewModel = viewModel;

        binding.setVariable(variable, viewModel);
    }

    public TViewModel getViewModel() {
        return viewModel;
    }
}

File: kotlin/app/src/main/java/com/pgssoft/kotlinplayground/view/adapter/BindingViewHolder.kt
package com.pgssoft.kotlinplayground.view.adapter

import android.databinding.ViewDataBinding
import android.support.v7.widget.RecyclerView
import com.pgssoft.kotlinplayground.viewmodel.BaseListItemViewModel

/**
 * Created by bstokrocki on 07.02.2017.
 */

class BindingViewHolder<TData, TViewModel : BaseListItemViewModel<TData>>(
        val viewModel: TViewModel,
        binding: ViewDataBinding,
        variable: Int) : RecyclerView.ViewHolder(binding.root) {

    init {
        binding.setVariable(variable, viewModel)
    }
}
